sources/datasources/clubs.js

```diff
@@ -90,15 +92,29 @@ class ClubAPI extends DataSource {
 				accessArray.map(async (accessItem, index) => {
 					const userId = accessItem.user;
 					const foundUser = await Users.findById(userId);
-					const accessObj = {
-						level: accessItem.level,
-						user: foundUser._id,
-						club: clubId,
-					};
-					let createdAccessLevel = await AccessLevel.create(accessObj);
-					updatedClub.memberAccess.push(createdAccessLevel);
-					foundUser.clubAccess.push(createdAccessLevel);
-					await foundUser.save();
+					const foundAccessObj=await AccessLevel.findOne({user:userId,club:foundClub._id});
+					//Check if there is no such access level defined
+					if(foundAccessObj==undefined){
```

Comment:
Is check for null not needed?
undefined and null are separate na

Reply:
It won't cause an issue because null==undefined

Reply:
I avoided use of === operator for this reason only

Reply:
No @DesignrKnight , the check for null is not needed because. if the object is not found, the `findOne` wont return anything, in that case the `foundAccessObj` would be undefined and not `null`